source/Ship.cpp

```diff
@@ -737,6 +739,8 @@ string Ship::FlightCheck() const
 		return "solar power?";
 	if(!hyperDrive && !jumpDrive && !canBeCarried)
 		return "no hyperdrive?";
+	if(!repair && !shield)
```

Comment:
This checks both hull and shield regen, and absent both, shows a "no shield gen" message. What about ships that have no shields?

Reply:
You mean, if a ship has hull repair but no shield generation? I'm not sure what to do in that case.

Reply:
No, I mean literally, if a ship has no shields at all, then a shield regenerator is a waste of space for it,  Unless the available content for the player includes hull repair tech, then they will have an unremovable warning symbol.